### ISSUE TITLE
Some errors aren't caught by the ErrorBoundary

### ISSUE THREAD

**Issue by denniskigen** [openmrs/openmrs-form-engine-lib#17]
## Steps to replicate

Attempt to open any form that uses the `subforms` feature e.g. the SOAP NOTE template form.

## Screenshot

<img width="1800" alt="Screenshot 2023-04-11 at 1 41 20 PM" src="https://user-images.githubusercontent.com/8509731/231137176-0c63a9b5-586d-4ca3-80f1-aced69cdcef6.png">

**Comment by samuelmale** [comment]
I agree, we should have an error boundary at the root level of a form; as for the root cause of the "subform" issue, I'm actively looking into that as of today. I'm happy with moving this forward :)

**Comment by denniskigen** [comment]
This should be fixed. Can you confirm, @samuelmale?
